node_modules/oae-admin/downloadlog/js/downloadlog.js

```diff
@@ -0,0 +1,227 @@
+                            errorMessage = '__MSG__GET_FAIL_NO_USERS__';
+                        } else {
+                            errorMessage = '__MSG__GET_FAIL_CONTENT__';
+                        }
+
+                        // Show an error notification
+                        oae.api.util.notification(
+                            oae.api.i18n.translate('__MSG__GET_FAIL__', 'downloadlog'),
+                            oae.api.i18n.translate(errorMessage, 'downloadlog'),
+                        'error');
+                    } else {
+                        // Show an error notification
+                        oae.api.util.notification(
+                            oae.api.i18n.translate('__MSG__GET_SUCCESS__', 'downloadlog'),
+                            oae.api.i18n.translate('__MSG__GET_SUCCESS_CONTENT__', 'downloadlog')
+                        );
+                        downloadCSV(log, 'logs_date-last-login_' + date + '.csv');
+                    }
+                });
+            }
+            // If the admin wants the creation date of each user
+            if (creationDate) {
+                oae.api.admin.getUsersDateLog(null, tenantAlias, 'created', function(err, log) {
+                    if (err) {
+                        // Show an error notification
+                        oae.api.util.notification(
+                            oae.api.i18n.translate('__MSG__GET_FAIL__', 'downloadlog'),
+                            oae.api.i18n.translate('__MSG__GET_FAIL_CONTENT__', 'downloadlog'),
+                        'error');
+                    } else {
+                        // Show an error notification
+                        oae.api.util.notification(
+                            oae.api.i18n.translate('__MSG__GET_SUCCESS__', 'downloadlog'),
+                            oae.api.i18n.translate('__MSG__GET_SUCCESS_CONTENT__', 'downloadlog')
+                        );
+                        downloadCSV(log, 'logs_date-creation_' + date + '.csv');
+                    }
+                });
+            }
+            return callback(null);
+        };
+
+        /**
+         * Close the modal
+         */
+        var closeModal = function() {
+            $('#downloadlog-modal', $rootel).modal('hide');
+        };
+
+        /**
+         * Download logins file 
+         *
+         * @param  {Object}     data            Data to download
+         * @param  {String}     fileName        File name
+         */
+        var downloadCSV = function(data, fileName) {
+            var element = document.createElement('a');
+            element.setAttribute('href', 'data:text/csv;charset=utf-8,' + encodeURIComponent(data));
+            element.setAttribute('download', fileName);
+
+            element.style.display = 'none';
+            document.body.appendChild(element);
+
+            element.click();
+
+            document.body.removeChild(element);
+        };
```

Comment:
there has to be a better way of doing this ☝️ creating an invisible DOM element and then clicking it is such a hack